snapcraft/internal/os_release.py

```diff
@@ -0,0 +1,89 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+
+from snapcraft.internal import errors
+
+_ID_TO_UBUNTU_CODENAME = {
+    '17.10': 'artful',
+    '17.04': 'zesty',
+    '16.04': 'xenial',
+    '14.04': 'trusty',
+}
+
+
+class OsRelease:
+    """A class to intelligently determine the OS on which we're running"""
+
+    def __init__(self, *, os_release_file='/etc/os-release'):
+        """Create a new OsRelease instance.
+
+        :param str os_release_file: Path to os-release file to be parsed.
+        """
+        with open(os_release_file) as os_release_file:
```

Comment:
Just a nit about reusing variables here. I would prefer to name one os_release_file_path or os_release_path, and the other os_release_file.

Reply:
Oh, yeah I agree, not sure what I was thinking there!